staging/src/kubesphere.io/api/types/v1beta1/federateduser_types.go

```diff
@@ -45,7 +45,10 @@ type FederatedUserSpec struct {
 }
 
 type UserTemplate struct {
-	Spec v1alpha2.UserSpec `json:"spec,omitempty"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              v1alpha2.UserSpec `json:"spec"`
+	// +optional
+	Status v1alpha2.UserStatus `json:"status,omitempty"`
```

Comment:
@wansir  There are a lot of code refactoring in the commits which seems doesn't relate to the root cause.  Could summarize how does the `User` reconcile be skipped in the Member cluster?

Reply:
I've refactored the user controller with controller runtime, controller runtime is simpler than resource client. The federated types defined in `federateduser_types.go` is incorrect.

Reply:
The user label will not be synchronized to the member cluster, but the user controller in the member cluster will modify the user status according to the incorrect label. In fact, the user status is controlled by the user controller in the host cluster.